pinot-spi/src/main/java/org/apache/pinot/spi/ingestion/batch/IngestionJobLauncher.java

```diff
@@ -136,23 +137,33 @@ public static void runIngestionJob(SegmentGenerationJobSpec spec) {
     }
   }
 
-  private static void kickoffIngestionJob(SegmentGenerationJobSpec spec, String ingestionJobRunnerClassName) {
-    LOGGER.info("Trying to create instance for class {}", ingestionJobRunnerClassName);
+  private static void kickoffIngestionJob(SegmentGenerationJobSpec spec, JobRunner jobRunner) {
+    String jobRunnerClassName = jobRunner.getClassName();
+    LOGGER.info("Trying to create instance for class {}", jobRunnerClassName);
     IngestionJobRunner ingestionJobRunner;
     try {
-      ingestionJobRunner = PluginManager.get().createInstance(ingestionJobRunnerClassName);
+      Preconditions.checkState(
+              jobRunnerClassName != null,
+              jobRunner.getPropertyName() + " in job spec was null"
+      );
+      ingestionJobRunner = PluginManager.get().createInstance(jobRunnerClassName);
     } catch (Exception e) {
       throw new RuntimeException(
-          "Failed to create IngestionJobRunner instance for class - " + ingestionJobRunnerClassName, e);
+              "Failed to create IngestionJobRunner instance for class - " + jobRunnerClassName, e);
     }
     ingestionJobRunner.init(spec);
     try {
       ingestionJobRunner.run();
     } catch (Exception e) {
-      throw new RuntimeException("Caught exception during running - " + ingestionJobRunnerClassName, e);
+      throw new RuntimeException("Caught exception during running - " + jobRunnerClassName, e);
     }
   }
 
+  public interface JobRunner {
```

Comment:
This abstraction doesn't seem clean to me (it is not really a job runner). Can we move the pre-condition check into the getters in `ExecutionFrameworkSpec`? Or we can add checked getters such as `getSegmentGenerationJobRunnerClassNameNotNull()`

Reply:
Oh yeh, that's a way better idea! I've updated it to use the `NotNull` suffix.